src/App.tsx

```diff
@@ -25,110 +25,38 @@ function Game() {
     'd',
   ]);
 
-  const handleKeyDown = (e: KeyboardEvent) => {
-    if (!moving) {
-      if (validKeys.has(e.key)) {
-        setMoving(true);
-        console.log('Key pressed: ' + e.key);
-        if (e.key === 'ArrowUp' || e.key === 'w') {
-          setDirection('n');
-        } else if (e.key === 'ArrowDown' || e.key === 's') {
-          setDirection('s');
-        } else if (e.key === 'ArrowRight' || e.key === 'd') {
-          setDirection('e');
-        } else if (e.key === 'ArrowLeft' || e.key === 'a') {
-          setDirection('w');
-        }
-      }
-    } else {
-      if (
-        validKeys.has(e.key) &&
-        !(direction === 'n' && (e.key === 'w' || e.key === 'ArrowUp')) &&
-        !(direction === 's' && (e.key === 's' || e.key === 'ArrowDown')) &&
-        !(direction === 'e' && (e.key === 'd' || e.key === 'ArrowRight')) &&
-        !(direction === 'w' && (e.key === 'a' || e.key === 'ArrowLeft')) &&
-        !(
-          direction === 'ne' ||
-          direction === 'nw' ||
-          direction === 'se' ||
-          direction === 'sw'
-        )
-      ) {
-        console.log(
-          'Already moving: ' + direction + '; new key pressed: ' + e.key
-        );
-        if (direction === 'n') {
-          if (e.key === 'ArrowRight' || e.key === 'd') {
-            setDirection('ne');
-          } else if (e.key === 'ArrowLeft' || e.key === 'a') {
-            setDirection('nw');
-          } else if (e.key === 'ArrowDown' || e.key === 's') {
-            setMoving(false);
-            setDirection('');
-          }
-        } else if (direction === 'e') {
-          if (e.key === 'ArrowUp' || e.key === 'w') {
-            setDirection('ne');
-          } else if (e.key === 'ArrowDown' || e.key === 's') {
-            setDirection('se');
-          } else if (e.key === 'ArrowLeft' || e.key === 'a') {
-            setMoving(false);
-            setDirection('');
-          }
-        } else if (direction === 's') {
-          if (e.key === 'ArrowLeft' || e.key === 'a') {
-            setDirection('sw');
-          } else if (e.key === 'ArrowRight' || e.key === 'd') {
-            setDirection('se');
-          } else if (e.key === 'ArrowUp' || e.key === 'w') {
-            setMoving(false);
-            setDirection('');
-          }
-        } else if (direction === 'w') {
-          if (e.key === 'ArrowDown' || e.key === 's') {
-            setDirection('sw');
-          } else if (e.key === 'ArrowUp' || e.key === 'w') {
-            setDirection('nw');
-          } else if (e.key === 'ArrowRight' || e.key === 'd') {
-            setMoving(false);
-            setDirection('');
-          }
-        }
-      }
+const handleKeyDown = (e: KeyboardEvent) => {
+    if (validKeys.includes(e.key)) {
```

Comment:
issue: after updating the branch w master, `validKeys` is a set, so `.includes()` will error and not function - i'll adjust and commit